src/Microsoft.Crank.Models/Job.cs

```diff
@@ -281,6 +281,14 @@ public Source Source
         /// </summary>
         public bool PatchReferences { get; set; } = false;
 
+        public Dictionary<string, List<CommandDefinition>> Commands { get; set; }
+        
+        public List<string> PreCommandOrder { get; set; } = new();
```

Comment:
I am wondering if these should have arguments. After all the scripts could read arguments. These could be a list of name/value properties on each command. So the string[] becomes a Command[] with `{ name, arguments[] { name, value }`

Reply:
I've made the name change to `BeforeJob` and `AfterJob`. I was thinking about how it would look to add arguments to the commands, but since the commands support variable substitution, I think that variable substitution feels more natural as the approach to parameterisation here. I know at least in my use case for this feature I plan on using > 15 variables and it'd be tedious to have to define all those arguments I think.